node/src/chain/acceptor.rs

```diff
@@ -171,25 +176,85 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
         Ok(())
     }
 
-    fn needs_update(blk: &Block, txs: &[SpentTransaction]) -> bool {
+    fn selective_update(
+        blk: &Block,
+        txs: &[SpentTransaction],
+        vm: &tokio::sync::RwLockWriteGuard<'_, VM>,
+        provisioners_list: &mut tokio::sync::RwLockWriteGuard<
+            '_,
+            ContextProvisioners,
+        >,
+    ) -> Result<()> {
+        // FIXME: Change `changed_provisioners` to return the status of changes.
+        // That way, we can handle everything here without asking for
+        // `get_provisioner`
+        let changed_prov = Self::changed_provisioners(blk, txs)?;
```

Comment:
Can this be done in this PR?

Reply:
My first goal was to have a mitigation patch.
Let see if I can handle it now

Reply:
Partially done.
Removing the `get_provisioner` would mean to duplicate the stake_contract logic in the binary.
What I did instead, is to handle the `stake` tx so to remove the the need for `get_provisioners` when epoch change.
Furthermore, having the status of the change, helped in writing better asserts